bin/main.rb

```diff
@@ -1,94 +1,89 @@
-    # check if player has made a win move,
-    if moves_counter > 4
-      puts "#{first_player} has won! congrats!!" + '--> if there\'s a winning move, of course, now breaks'
-      # somebody_won = true
-      # breaks the loop
-      # check if have been reached 9 moves and no winner,
-      puts 'there\'s been a tie' + '--> if there is a tie, of course, now breaks'
-      # break the  loop
-    end
-    puts ' '
+# links to classes inside /lib
+
+require_relative '../lib/board'
+require_relative '../lib/player'
+
+# Introduce the game to the players
+
+def cover_image
+  puts
+  puts '           ▛┅┅┅┳┅┅┅┳┅┅▜'
+  puts '           ┇ 1 ┇ 2 ┇ 3 ┇'
+  puts '           ┣┅┅┅╋┅┅┅╋┅┅┅┫'
+  puts '           ┇ 4 ┇ 5 ┇ 6 ┇'
+  puts '           ┣┅┅┅╋┅┅┅╋┅┅┅┫'
+  puts '           ┇ 7 ┇ 8 ┇ 9 ┇'
+  puts '           ▙┅┅┅┻┅┅┅┻┅┅▟'
+  puts
+  puts 'Welcome to TIC-TAC-TOE by maosan132'
+  puts
+end
+
+game_on = true
+new_board = Board.new
+count = 0
+
+def update_screen
+  Gem.win_platform? ? (system 'cls') : (system 'clear')
+  cover_image
+end
+
+update_screen
+print 'Player one, please type your name: '
+player_name = gets.chomp
+player_one = Player.new(player_name)
+puts "Welcome, #{player_one.name}. The X is yours"
+
+print 'Player two, please type your name: '
+player_name = gets.chomp
+player_two = Player.new(player_name)
+puts "Welcome, #{player_two.name}. The O is yours"
+
+while game_on
+  update_screen
+  new_board.display
+  count += 1
+
+  if count.odd?
+    turn = 'X'
+    player = player_one.name
   else
-    puts "#{second_player}, make a choice"
-    # Show available numbers from the updated array of options
-    selection = gets.chomp
-    selection = selection.to_i
-    # make mentioned checks, if ok, assings value to cell position else repeat
-    # remove number choice from array of options
-    # if player has made a win move,
-    if moves_counter > 4
-      puts "#{second_player} has won! congrats!!" + '--> if there is a winning move, of course, now breaks'
-      # somebody_won = true
-      # breaks the loop
-      # check if have been reached 9 moves and no winner,
-      puts 'there is been a tie' + '--> if there\'s a tie, of course, now breaks'
-      # break the  loop
-    end
-    puts
+    turn = 'O'
+    player = player_two.name
+  end
+  puts "turn is #{turn}"
+  print "#{player}, Choose a cell number from 1 to 9: "
+  cell_choice = gets.chomp.to_i
+
+  valid = new_board.valid(cell_choice)
+  unless valid
+    print "#{turn}, Choose a cell number from 1 to 9: "
+    cell_choice = gets.chomp.to_i
   end
-  # displays the updated board
-  puts '<-----Draws the updated board----->'
+
+  taken = new_board.taken?(cell_choice)
+  if taken
+    print 'that number is not available, choose other one: '
+    cell_choice = gets.chomp.to_i
+  end
+  puts
+
+  new_board.update(cell_choice, turn)
   puts
-  moves_counter += 1
+
+  if new_board.horizontal_winner?(turn) || new_board.vertical_winner?(turn) || new_board.diagonal_winner?(turn)
```

Comment:
Write a method `win?` and replace this line with 
```rb
if board.win?(turn)
    ...
end
```
It's soft, cleaner, more readable.

Reply:
done, was good advice. I'll keep it present